app-backend/db/src/main/scala/SceneToLayerDao.scala

```diff
@@ -0,0 +1,239 @@
+    )
+    val select = fr"""
+    SELECT
+      scene_id, project_id, project_layer_id, accepted, scene_order, mosaic_definition, scene_type, ingest_location,
+      data_footprint, is_single_band, single_band_options
+    FROM (
+      scenes_to_layers
+    LEFT JOIN
+      scenes
+    ON scenes.id = scenes_to_layers.scene_id
+    ) scenes_stl
+    LEFT JOIN
+      project_layers
+    ON
+      scenes_stl.project_layer_id = project_layers.id
+    LEFT JOIN
+      projects
+    ON
+      project_layers.project_id = projects.id
+      """
+    (select ++ whereAndOpt(filters: _*) ++ fr"ORDER BY scene_order ASC")
+      .query[SceneToLayerWithSceneType]
+      .stream map { stp =>
+      {
+        Applicative[Option].map3(redBand, greenBand, blueBand) {
+          case (r, g, b) =>
+            MosaicDefinition(
+              stp.sceneId,
+              stp.colorCorrectParams.copy(
+                redBand = r,
+                greenBand = g,
+                blueBand = b
+              ),
+              stp.sceneType,
+              stp.ingestLocation,
+              stp.dataFootprint flatMap { _.geom.as[MultiPolygon] },
+              stp.isSingleBand,
+              stp.singleBandOptions
+            )
+        } getOrElse {
+          MosaicDefinition(
+            stp.sceneId,
+            stp.colorCorrectParams,
+            stp.sceneType,
+            stp.ingestLocation,
+            stp.dataFootprint flatMap { _.geom.as[MultiPolygon] },
+            stp.isSingleBand,
+            stp.singleBandOptions
+          )
+        }
+      }
+    }
+  }
+
+  def getMosaicDefinition(
+      projectLayerId: UUID): Stream[ConnectionIO, MosaicDefinition] = {
+    getMosaicDefinition(projectLayerId, None)
+  }
+
+  def getColorCorrectParams(
+      projectLayerId: UUID,
+      sceneId: UUID): ConnectionIO[ColorCorrect.Params] = {
+    query
+      .filter(
+        fr"project_layer_id = ${projectLayerId} AND scene_id = ${sceneId}")
+      .select
+      .map { stl: SceneToLayer =>
+        stl.colorCorrectParams
+      }
+  }
+
+  def setColorCorrectParams(
+      projectLayerId: UUID,
+      sceneId: UUID,
+      colorCorrectParams: ColorCorrect.Params): ConnectionIO[SceneToLayer] = {
+    fr"""
+      UPDATE scenes_to_layers
+      SET mosaic_definition = ${colorCorrectParams}
+      WHERE project_layer_id = ${projectLayerId} AND scene_id = ${sceneId}
+    """.update.withUniqueGeneratedKeys("scene_id",
+                                       "project_layer_id",
+                                       "accepted",
+                                       "scene_order",
+                                       "mosaic_definition")
+  }
+
+  def setColorCorrectParamsBatch(
+      projectLayerId: UUID,
+      batchParams: BatchParams): ConnectionIO[List[SceneToLayer]] = {
+    batchParams.items
+      .map(params =>
+        setColorCorrectParams(projectLayerId, params.sceneId, params.params))
+      .sequence
+  }
+
+  def setProjectLayerColorBands(
```

Comment:
What is this function doing? I'm a little confused -- if there is a mosaic definition it's keeping it, if not it's updating the color bands?

Also - seems a little risky (I know we're serializing it into a `case class` so maybe not) to take take user input and use `Fragment.const` without escaping here).

Reply:
This function is being called in `projects/<id>/project-color-mode` endpoint, which is a `POST` endpoint, and which is being used on frontend when changing project color mode with `ProjectColorModeParams` (just RGB band numbers). 

`mosaic_definition ` is a `jsonb` field, so `||` operator, I think, will do concatenation. And if we `jsonb '{"a":1}' || jsonb '{"a":2}'`, it will yield ` {"a": 2}`.

I agree, I will use `fr""` instead for those user inputs.